services/src/main/java/org/keycloak/authentication/AuthenticationProcessor.java

```diff
@@ -1015,13 +1014,15 @@ public static ClientSessionContext attachSession(AuthenticationSessionModel auth
                 userSession.restartSession(realm, authSession.getAuthenticatedUser(), username, connection.getRemoteAddr(), authSession.getProtocol()
                         , remember, brokerSessionId, brokerUserId);
             } else {
-                // We have existing userSession even if it wasn't attached to authenticator. Could happen if SSO authentication was ignored (eg. prompt=login) and in some other cases.
-                // We need to handle case when different user was used
-                logger.debugf("No SSO login, but found existing userSession with ID '%s' after finished authentication.", userSession.getId());
                 if (!authSession.getAuthenticatedUser().equals(userSession.getUser())) {
-                    event.detail(Details.EXISTING_USER, userSession.getUser().getId());
-                    event.error(Errors.DIFFERENT_USER_AUTHENTICATED);
-                    throw new ErrorPageException(session, authSession, Response.Status.INTERNAL_SERVER_ERROR, Messages.DIFFERENT_USER_AUTHENTICATED, userSession.getUser().getUsername());
+                    AuthenticationManager.backchannelLogout(session, userSession, true);
```

Comment:
@mposolda Thank you for the review and suggestions. I'm not sure about this particular change. I removed the original branch where an error was thrown. Maybe we need to somehow check from what "state" we ended up here where we have one active user session but different user in authSession. By "state" I mean for example we came here from registration form and we want to log in different user or the original user was disabled and we want to log in different user. 
The bottom line: I am not sure whether we can always log out original user and create new authSession without checking why this situation happened.

Reply:
Well, my hope is, we can always logout original user and create new userSession. This can be for all flows like "register", "reset-credentials" or "login" and will allow to avoid some additional complexity.

BTV. There is discussion on keycloak-dev regarding this, so I suggest to continue discussion there.